docs/tutorials/other/rust.rst

```diff
@@ -0,0 +1,275 @@
+    :emphasize-lines: 3-7, 13-16
+
+    #![no_std]
+
+    extern "C" {
+        fn sysinit_start();
+        fn sysinit_app();
+        fn sysinit_end();
+    }
+
+    extern crate panic_halt;
+
+    #[no_mangle]
+    pub extern "C" fn main() {
+        /* Initialize all packages. */
+        unsafe { sysinit_start(); }
+        unsafe { sysinit_app(); }
+        unsafe { sysinit_end(); }
+
+        loop {
+        }
+    }
+
+First we manually define the three sysinit functions. This is similar to the
+C header file. Then we execute the sysinit as the macro would do. 
+
+We need the ``unsafe`` indication because the C code doesn't have the same memory
+guarantees as Rust. Normally we need to build a safe Rust wrapper, but that is 
+out of scope for this tutorial.
+
+Doing GPIO and delays
+~~~~~~~~~~~~~~~~~~~~~
+
+Now it is time to do some GPIO and add a delay. Again we define the functions 
+and then use them in and around the loop. We need some constants that are 
+normally defined by the BSP or MCU. These constants need to move to a better 
+place later.
+
+.. code-block:: rust
+    :emphasize-lines: 7-9, 14, 16, 25, 28-29, 31-32
+
+    #![no_std]
+
+    extern "C" {
+        fn sysinit_start();
+        fn sysinit_app();
+        fn sysinit_end();
+        fn hal_gpio_init_out(pin: i32, val: i32) -> i32;
+        fn hal_gpio_toggle(pin: i32);
+        fn os_time_delay(osticks: u32);
+    }
+
+    extern crate panic_halt;
+
+    const OS_TICKS_PER_SEC: u32 = 128;
+
+    const LED_BLINK_PIN: i32 = 23;
+
+    #[no_mangle]
+    pub extern "C" fn main() {
+        /* Initialize all packages. */
+        unsafe { sysinit_start(); }
+        unsafe { sysinit_app(); }
+        unsafe { sysinit_end(); }
+
+        unsafe { hal_gpio_init_out(LED_BLINK_PIN, 1); }
+
+        loop {
+            /* Wait one second */
+            unsafe { os_time_delay(OS_TICKS_PER_SEC); }
+
+            /* Toggle the LED */
+            unsafe { hal_gpio_toggle(LED_BLINK_PIN); }
+        }
+    }
+
+Cargo build
+~~~~~~~~~~~
+
+Now that the application is converted we need to build it and link it the rest of mynewt. We start with Cargo.toml:
+
+.. code-block:: toml
+    :emphasize-lines: 7-8, 10-11
+
+    [package]
+    name = "rust-klok"
+    version = "0.1.0"
+    authors = ["Casper Meijn <casper@meijn.net>"]
+    edition = "2018"
+
+    [dependencies]
+    panic-halt = "0.2.0"
+
+    [lib]
+    crate-type = ["staticlib"]
+
+This adds the ``panic-halt`` dependency, which is needed for the panic handler as 
+mentioned earlier. It also configures the crate as staticlib, which causes the
+application to be build as .a-library. This will be needed in a later step.
+
+Next we need a script for running ``cargo`` and moving the library to the correct 
+place. Create a new file named ``apps/blinky/cargo_build.sh`` with the following contents:
+
+.. code-block:: bash
+    :emphasize-lines: 1-5
+
+    #!/bin/bash
+    set -eu
+    TARGET="thumbv7m-none-eabi"
```

Comment:
Mynewt also supports many Cortex-M0 targets which are `thumbv6m-none-eabi`, not sure this would work in that case!